scripts/utils/formatNumber.js

```diff
@@ -0,0 +1,19 @@
+export default (number, isPercentage)  => {
+  let parsedNumber;
+
+  if (number === null || number === 'undefined' || typeof number !== 'number') return 'N/A';
+
+  if (number === 0) return number;
+
+  if (isPercentage) {
+    parsedNumber = (number * 100).toFixed(1);
+  } else {
+    if (number > 0 && number < 1) {
+      parsedNumber = number.toFixed(4);
+    } else {
+      parsedNumber = number.toFixed(2);
+    }
+  }
+
+  return parseFloat(parsedNumber).toLocaleString();
+};
```

Comment:
FYI, instead of doing 

`parseFloat(number.toFixed(4)).toLocaleString()`

It's also possible to do:

`number.toLocaleString(undefined, {
  minimumFractionDigits: 0,
  maximumFractionDigits: 4
});
`

It's longer but you have more control over the number of decimals.